provider/test-programs/regress-3835/index.ts

```diff
@@ -0,0 +1,45 @@
+import * as aws from "@pulumi/aws";
+
+const proxyRole = new aws.iam.Role(
+  "role",
+  {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal(
+      aws.iam.Principals.RdsPrincipal,
+    ),
+    managedPolicyArns: [],
+  },
+  {},
+);
+
+const vpc = new aws.ec2.Vpc("vpc", {
+  cidrBlock: "10.0.0.0/16",
+});
+
+const subnet = new aws.ec2.Subnet("subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.1.0/24",
+  availabilityZone: "us-west-2a",
+});
+
+const anotherSubnet = new aws.ec2.Subnet("another-subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.2.0/24",
+  availabilityZone: "us-west-2b",
+});
+
+const example = new aws.rds.Proxy("example", {
+  name: "example",
+  debugLogging: false,
+  engineFamily: "MYSQL",
+  idleClientTimeout: 1800,
+  requireTls: true,
+  roleArn: proxyRole.arn,
+  vpcSubnetIds: [subnet.id, anotherSubnet.id],
+  auths: [{}],
```

Comment:
It looks like the integration test is failing because this actually is invalid. If you want to actually run an `up` instead of just `preview` you can change this to

```ts
const secret = new aws.secretsmanager.Secret("secret", {});
const example = new aws.rds.Proxy("example", {
  debugLogging: false,
  engineFamily: "MYSQL",
  idleClientTimeout: 1800,
  requireTls: true,
  roleArn: proxyRole.arn,
  vpcSubnetIds: [subnet.id, anotherSubnet.id],
  auths: pulumi
    .all([secret.arn])
    .apply((sec): aws.types.input.rds.ProxyAuth[] => {
      return [
        {
          iamAuth: "DISABLED",
          authScheme: "SECRETS",
          secretArn: sec[0],
        },
      ];
    }),
  tags: {
    Name: "example",
    Key: "value",
  },
});
```

Although if we ever fix https://github.com/pulumi/pulumi-terraform-bridge/pull/1881 then we would not reproduce it with this test.

Reply:
Yes let's try the test with an unknown like you suggest. This regression test will probably be short-lived but I just wanted some confirmation that the patch edit is working.